### PR TITLE
Improve install script

### PR DESCRIPTION
**Pull request by aparkerlue**
Problem Statement
-----------------

Install script is (1) difficult to read, (2) does not print critical information about what it is doing, (3) silently modifies `~/.bash_profile`, and (4) writes files to multiple locations in the filesystem, a change that is difficult to revert unless there is a corresponding uninstallation mechanism.

Solution
--------

1. Update the install script to make it easier to read using here documents
2. Print information about what it is doing
3. Print advice about how to modify `~/.bash_profile` and how to proceed post-installation
4. Write all files to `~/.okta` so that uninstalling only involves removing that directory

### CODE REVIEW

**Comment by aparkerlue** [comment]
This is how `~/.okta` might look after installing using these changes and then authenticating:

```
$ tree -a ~/.okta/
~/.okta/
├── .current-session
├── bash_functions
├── bin
│   ├── okta-credential_process
│   ├── okta-listroles
│   └── withokta
├── config.properties
├── cookies.properties
├── fish_functions
│   ├── okta-aws.fish
│   └── okta-sls.fish
├── logging.properties
├── okta-aws-cli-1.0.7.jar
├── okta-aws-cli.jar -> okta-aws-cli-1.0.7.jar
└── profiles

2 directories, 13 files
```

File: bin/install.sh [hunk 1]
```diff
@@ -89,34 +101,44 @@ env OKTA_PROFILE=$profile java \
...
+EOF
+chmod +x "${dotokta}/bin/okta-listroles"
 
 # Configure Okta AWS CLI
-oktaConfig="${HOME}/.okta/config.properties"
-grep '^#OktaAWSCLI' "${oktaConfig}" > /dev/null 2>&1
-if [ $? -ne 0 ]
-then
-echo "
+oktaConfig="${dotokta}/config.properties"
```
Comment (AlainODea) [inline bin/install.sh]: This won't work unless PREFIX is ~/.okta/ which complicates system installs.

The location of config.properties is effectively hard-coded here:
https://github.com/oktadeveloper/okta-aws-cli-assume-role/blob/master/src/main/java/com/okta/tools/OktaAwsConfig.java#L96-L108

suggestion:
```suggestion
oktaConfig="${HOME}/.okta/config.properties"
```

**Comment by aparkerlue** [comment]
I'm about to commit a couple additional edits. I'll follow up.

**Comment by aparkerlue** [comment]
Usage examples:

### Invocation without arguments

```
$ ./install.sh 
usage: install.sh [-h | -i]
       install Okta AWS CLI Assume Role tool
```

### Help

```
$ ./install.sh -h
Installation script for Okta AWS CLI Assume Role
================================================

To execute:

    install.sh -i

This command

1. Installs files into a filesystem location that can be configured
   with the PREFIX environment variable (default: ~/.okta) and
2. Prints instructions for setting up shell functions and scripts.

This script checks for (and installs if necessary) the file
~/.okta/config.properties regardless of the value of PREFIX.

For details, see https://github.com/oktadeveloper/okta-aws-cli-assume-role.
```

### Installation into default location

```
$ ./install.sh -i
Installing into ~/.okta
Latest release JAR file: https://github.com/oktadeveloper/okta-aws-cli-assume-role/releases/download/v1.0.7/okta-aws-cli-1.0.7.jar
Fetching JAR file → ~/.okta/okta-aws-cli-1.0.7.jar
Symlinking ~/.okta/okta-aws-cli.jar → okta-aws-cli-1.0.7.jar
Found ~/.okta/config.properties

Add the following to ~/.bash_profile or ~/.profile:

#OktaAWSCLI
if [[ -f "$HOME/.okta/bash_functions" ]]; then
    . "$HOME/.okta/bash_functions"
fi
if [[ -d "$HOME/.okta/bin" && ":$PATH:" != *":$HOME/.okta/bin:"* ]]; then
    PATH="$HOME/.okta/bin:$PATH"
fi
```

### Installation into custom location

```
$ PREFIX=tmp ./install.sh -i
Installing into ~/tmp
Latest release JAR file: https://github.com/oktadeveloper/okta-aws-cli-assume-role/releases/download/v1.0.7/okta-aws-cli-1.0.7.jar
Fetching JAR file → ~/tmp/okta-aws-cli-1.0.7.jar
Symlinking ~/tmp/okta-aws-cli.jar → okta-aws-cli-1.0.7.jar
Found ~/.okta/config.properties

Add the following to ~/.bash_profile or ~/.profile:

#OktaAWSCLI
if [[ -f "$HOME/tmp/bash_functions" ]]; then
    . "$HOME/tmp/bash_functions"
fi
if [[ -d "$HOME/tmp/bin" && ":$PATH:" != *":$HOME/tmp/bin:"* ]]; then
    PATH="$HOME/tmp/bin:$PATH"
fi
```

**Comment by aparkerlue** [comment]
I modified the installation script to require `-i` to initiate the installation, so the invocation now looks like this:

```bash
bash <(curl -fsSL https://raw.githubusercontent.com/oktadeveloper/okta-aws-cli-assume-role/master/bin/install.sh) -i
```

It would be best to use a raw Github user content URL for the latest release.

**Comment by AlainODea** [comment]
@aparkerlue thank you for your contribution. I have reviewed it and tested it. It is a big improvement in code and UX and it works.

**Pull request merged**